### PR TITLE
kubeadm: Don't drain and remove the current node on kubeadm reset

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

In v1.5, `kubeadm reset` would drain your node and remove it from your cluster if you specified, but now in v1.6 we can't do that due to the RBAC rules we have set up.

After conversations with @liggitt, I also agree this functionality was somehow a little mis-placed (though still very convenient to use), so we're removing it for v1.6.

It's the system administrator's duty to drain and remove nodes from the cluster, not the nodes' responsibility.

The current behavior is therefore a bug that needs to be fixed in v1.6

**Release note**:

```release-note
kubeadm: `kubeadm reset` won't drain and remove the current node anymore
```
@liggitt @deads2k @jbeda @dmmcquay @pires @errordeveloper

### CODE REVIEW

**Comment by luxas** [comment]
Sorry, my intention was not to unassign @pires, that was a race condition in the Github page

**Comment by luxas** [comment]
@k8s-bot unit test this

**Review by pires — APPROVED** [review]
/lgtm

File: cmd/kubeadm/app/cmd/reset.go [hunk 1]
```diff
@@ -83,20 +76,13 @@ func NewReset(skipPreFlight, removeNode bool, certsDir string) (*Reset, error) {
...
-		certsDir:   certsDir,
+		certsDir: certsDir,
 	}, nil
 }
 
 // Run reverts any changes made to this host by "kubeadm init" or "kubeadm join".
 func (r *Reset) Run(out io.Writer) error {
 
-	// Try to drain and remove the node from the cluster
-	err := drainAndRemoveNode(r.removeNode)
```
Comment (liggitt) [inline cmd/kubeadm/app/cmd/reset.go]: would you want to mark it unschedulable or taint it instead? seems like it should do *something* to signal that no more pods should be sent

Comment (luxas) [inline cmd/kubeadm/app/cmd/reset.go]: I could do it, but I think the node will be marked NotReady by controller manager by default, right?
If we have to set the NodeNotReady taint I'll do it, but I'd like to change the minimum amount of things in this PR

**Comment by jbeda** [comment]
Haven't reviewed the code but...

Ideally we would do the following:
* Detect if the node is drained.  If it is not, then error out with instructions on what to run with admin privs.
* Provide a `--force` flag for reseting the node without draining.

Not sure we can bring this together in time for 1.6.

**Comment by luxas** [comment]
@jbeda I think we should do that long-term, but it won't fit into this release.
I think this is the best we can do, just revert to the earlier behavior.

**Comment by liggitt** [comment]
If this was previous behavior, I'm ok with reverting to it. admins *should* drain their node first, but if they don't, the nodecontroller will take care of the node when it stops updating status.

**Comment by jbeda** [comment]
Not excited about this as it is a regression in functionality but I guess it is the best we can do on short notice.

/lgtm

**Comment by luxas** [comment]
> Not excited about this as it is a regression in functionality but I guess it is the best we can do on short notice.

Feel exactly the same but this is better than erroring in 100 % of the cases.

Root cause it (as discussed) that with RBAC a node can't drain itself.
We should document the best practices around removing a node in the docs though. @lukemarsden

**Comment by luxas** [comment]
But thanks for the LGTM!

**Pull request merged**